### PR TITLE
Historical humidity process year month city

### PR DESCRIPTION
**Pull request by Rea888**

### CODE REVIEW

File: app/Service/HistoricalWeatherHumidityService.php [hunk 1]
```diff
@@ -24,4 +33,22 @@ public function processHistoricalWeatherHumidityApiResponse( array $apiResponse
```
Comment (KuveeTheBird) [inline app/Service/HistoricalWeatherHumidityService.php:33]: i know it didn't change in this branch, but:
you instantiate a HistoricalWeatherHumidityModel, then you treat it as an array...
please change
`$historicalWeatherHumidity['longitude'] = $longitude; 
$historicalWeatherHumidity['latitude'] = $latitude;
$historicalWeatherHumidity['relative_humidity_2m'] = $humidity;
$historicalWeatherHumidity['date_time_of_measurement'] = $dateTimeOfMeasurement;`

to

`$historicalWeatherHumidity->longitude = $longitude; 
$historicalWeatherHumidity->latitude = $latitude;
$historicalWeatherHumidity->relative_humidity_2m = $humidity;
$historicalWeatherHumidity->date_time_of_measurement = $dateTimeOfMeasurement;`

Comment (KuveeTheBird) [reply app/Service/HistoricalWeatherHumidityService.php:33]: - [x] fix Model property usage

**Review by KuveeTheBird — CHANGES_REQUESTED** [review]

**Comment by Rea888** [comment]
fixed the required changes :)

File: app/Models/HistoricalHumidityProcessingReportsModel.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
```
Comment (KuveeTheBird) [inline app/Models/HistoricalHumidityProcessingReportsModel.php]: - [x] this file isn't part of the current task/branch, pls remove

File: database/migrations/2023_02_16_094504_create_historical_humidity_processing_reports_models_table.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?php
```
Comment (KuveeTheBird) [inline database/migrations/2023_02_16_094504_create_historical_humidity_processing_reports_models_table.php]: - [x] this file isn't part of the current task/branch, pls remove

File: app/Repository/HistoricalWeatherTenYearDatesCreator.php [hunk 4]
```diff
@@ -40,4 +40,14 @@ public function endDatesArray(): array
         return $data;
     }
 
+    public function getCityYearMonthArray(): array
+    {
+        $cityYearMonthArray = array(
+
```
Comment (KuveeTheBird) [inline app/Repository/HistoricalWeatherTenYearDatesCreator.php]: - [x] remove unnecessary empty line

File: app/Service/HistoricalWeatherHumidityService.php [hunk 5]
```diff
@@ -3,25 +3,52 @@
...
+
+        $latitudeAndLongitude = $this->longitudeLatitudeRepository->getLatitudeAndLongitude($cityName);
+        $latitude = $latitudeAndLongitude->getLatitude();
+        $longitude = $latitudeAndLongitude->getLongitude();
+
+
+        $startDate = $year . '-' . $month . '-01';
+        $endDate = date("Y-m-t", strtotime($startDate));
+
+        $get_data = Http::get('https://archive-api.open-meteo.com/v1/archive?latitude=' . $latitude . '&longitude=' . $longitude . '&start_date=' . $startDate . '&end_date=' . $endDate . '&hourly=relativehumidity_2m');
```
Comment (KuveeTheBird) [inline app/Service/HistoricalWeatherHumidityService.php]:
```suggestion
        $apiResponse = Http::get('https://archive-api.open-meteo.com/v1/archive?latitude=' . $latitude . '&longitude=' . $longitude . '&start_date=' . $startDate . '&end_date=' . $endDate . '&hourly=relativehumidity_2m');
```

File: app/Service/HistoricalWeatherHumidityService.php [hunk 6]
```diff
@@ -3,25 +3,52 @@
...
+        $response = json_decode($get_data, true);
```
Comment (KuveeTheBird) [inline app/Service/HistoricalWeatherHumidityService.php]:
```suggestion
        $parsedResponse = json_decode($apiResponse, true);
```

File: app/Service/HistoricalWeatherHumidityService.php [hunk 7]
```diff
@@ -3,25 +3,52 @@
...
+        $this->processHistoricalWeatherHumidityApiResponse($response);
```
Comment (KuveeTheBird) [inline app/Service/HistoricalWeatherHumidityService.php]:
```suggestion
        $this->processHistoricalWeatherHumidityApiResponse($parsedResponse);
```

**Review by KuveeTheBird — CHANGES_REQUESTED** [review]

File: app/Repository/HistoricalWeatherTenYearDatesCreator.php [hunk 8]
```diff
@@ -40,4 +40,14 @@ public function endDatesArray(): array
         return $data;
     }
 
+    public function getCityYearMonthArray(): array
+    {
+        $cityYearMonthArray = array(
```
Comment (KuveeTheBird) [inline app/Repository/HistoricalWeatherTenYearDatesCreator.php]: - [x] switch to short array declaration syntax

**Review by KuveeTheBird — APPROVED** [review]

**Pull request merged**